TESTING.md

```diff
@@ -0,0 +1,34 @@
+# Running Tests
+
+## Prerequisites
+Before running tests, ensure all steps in the [BUILDING.md](https://github.com/apache/geode/blob/develop/BUILDING.md) have been performed.
+
+## Running Tests By Type
```

Comment:
Just looking at the other checks on GitHub, do we not care about AcceptanceTest, StressNewTest, or UpgradeTest?  StressNewTest has zero tests from a quick glance and the other two have corresponding `gradle` calls

Reply:
I think it makes sense to include acceptance and upgrade tests in the instructions.  I'll add them.  StressNewTest is really for CI - we "stress" brand new tests by running them many times to expose any unforeseen timing issues.  I don't think it belongs in these instructions.